rttest/src/rttest.cpp

```diff
@@ -500,8 +499,8 @@ int rttest_init(
 
 int Rttest::get_next_rusage(size_t i)
 {
-  size_t prev_maj_pagefaults = this->prev_usage.ru_majflt;
-  size_t prev_min_pagefaults = this->prev_usage.ru_minflt;
+  int64_t prev_maj_pagefaults = this->prev_usage.ru_majflt;
+  int64_t prev_min_pagefaults = this->prev_usage.ru_minflt;
```

Comment:
Technically, I don't think this is correct.  In particular, I think this is going to be too large on 32-bit systems, since the man page for `getrusage` says that this is a long:

```
       The resource usages are returned in the structure pointed to by usage, which has the following form:

           struct rusage {
               struct timeval ru_utime; /* user CPU time used */
               struct timeval ru_stime; /* system CPU time used */
               long   ru_maxrss;        /* maximum resident set size */
               long   ru_ixrss;         /* integral shared memory size */
               long   ru_idrss;         /* integral unshared data size */
               long   ru_isrss;         /* integral unshared stack size */
               long   ru_minflt;        /* page reclaims (soft page faults) */
               long   ru_majflt;        /* page faults (hard page faults) */
               long   ru_nswap;         /* swaps */
               long   ru_inblock;       /* block input operations */
               long   ru_oublock;       /* block output operations */
               long   ru_msgsnd;        /* IPC messages sent */
               long   ru_msgrcv;        /* IPC messages received */
               long   ru_nsignals;      /* signals received */
               long   ru_nvcsw;         /* voluntary context switches */
               long   ru_nivcsw;        /* involuntary context switches */
           };
```

So to be more correct, I think we want to mark these as `long`.

Reply:
Ah, I didn't consider 32-bit systems.

I initially tried to use `long`, but I get a lint failure when I run the tests: 
```
   -- run_test.py: invoking following command in '/root/ros2_ws/src/ros2/realtime_support/rttest':
     - /root/ros2_ws/install/ament_cpplint/bin/ament_cpplint --xunit-file /root/ros2_ws/build/rttest/test_results/rttest/cpplint.xunit.xml
    /root/ros2_ws/src/ros2/realtime_support/rttest/src/rttest.cpp:502:  Use int16/int64/etc, rather than the C type long  [runtime/int] [4]
    /root/ros2_ws/src/ros2/realtime_support/rttest/src/rttest.cpp:503:  Use int16/int64/etc, rather than the C type long  [runtime/int] [4]

```
Should I tell the linter to skip this line?

Reply:
Yeah, I think so.  I also typically like to use the `[u]int*` family to be clear on the size, but since we are dealing with an interface outside of our control, I think using long here is the right thing.  Just add a comment on why you are skipping this line with the linter.